### PR TITLE
add source functions from agenda to computation-graph

### PR DESCRIPTION
**Pull request by romyzilkha**

### CODE REVIEW

File: computation_graph/graph.py [hunk 1]
```diff
@@ -123,3 +124,46 @@ def unbound_signature(
...
+
+def _transform_edges(query, edge_mapper):
+    return _operate_on_subgraph(
+        _split_by_condition(query), gamla.compose(tuple, gamla.map(edge_mapper))
+    )
+
+
+def _edge_source_equals(x):
+    if not isinstance(x, base_types.ComputationNode):
+        x = make_computation_node(x)
```
Comment (romCohen) [inline computation_graph/graph.py:141]: This if statement is redundant, it's already done at the beginning of `make_computation_node`.
```
def _edge_source_equals(x):
    x = make_computation_node(x)
    return lambda edge: edge.source == x
```

File: computation_graph/graph.py [hunk 2]
```diff
@@ -123,3 +124,46 @@ def unbound_signature(
...
+    return lambda edge: edge.source == x
+
+
+def _replace_edge_source(replacement):
+    if not isinstance(replacement, base_types.ComputationNode):
```
Comment (romCohen) [inline computation_graph/graph.py:146]: Same here.

File: computation_graph/graph.py [hunk 3]
```diff
@@ -123,3 +124,46 @@ def unbound_signature(
...
+        replacement = make_computation_node(replacement)
+
+    def replace_edge_source(edge):
+        return dataclasses.replace(edge, source=replacement)
+
+    return replace_edge_source
```
Comment (romCohen) [inline computation_graph/graph.py:152]: In `_edge_source_equals` we retuned a `lambda` and here we are returning an inner function. I think we need to be consistent here, @uriva Which do you prefer? 
I personally prefer the inner function pattern.

**Review by romCohen — APPROVED** [review]

**Review by uriva — APPROVED** [review]

File: computation_graph/graph.py (hunk 1 above)
Comment (uriva) [reply computation_graph/graph.py:141]: I think we use it for both types, but maybe we can change it.

File: computation_graph/graph.py (hunk 3 above)
Comment (uriva) [reply computation_graph/graph.py:152]: probably an inner function or a composition

**Pull request merged**